ehrql/query_engines/base_sql.py

```diff
@@ -216,8 +216,16 @@ def get_sql_is_null(self, node):
     @get_sql.register(Function.In)
     def get_sql_in(self, node):
         lhs = self.get_expr(node.lhs)
-        rhs = self.get_expr_for_multivalued_param(node.rhs)
-        return lhs.in_(rhs)
+        if isinstance(node.rhs, Value) and len(node.rhs.value) == 0:
+            # Special case handling for when LHS is NULL and RHS is an empty list: ehrQL
+            # evaluates this as NULL whereas SQL evaluates it as FALSE. (Of course, both
+            # agree than when RHS is empty and LHS is non-NULL then the result is
+            # FALSE.) So we want an expression which evaluates NULL when LHS is NULL,
+            # and FALSE otherwise. The expression `LHS != LHS` behaves as required.
+            return operators.ne(lhs, lhs)
```

Comment:
Ah, we don't! But the point is that _when_ it's NULL we want the result to be NULL, and otherwise we want it to be FALSE. And that's what `LHS != LHS` gives us.